iso15118/secc/controller/ev_data.py

```diff
@@ -0,0 +1,187 @@
+    ev_target_energy_request: Optional[float] = None
+    ev_max_energy_request: Optional[float] = None
+    ev_min_energy_request: Optional[float] = None
+
+    ev_max_charge_power: Optional[float] = None
+    ev_max_charge_power_l2: Optional[float] = None
+    ev_max_charge_power_l3: Optional[float] = None
+
+    ev_min_charge_power: Optional[float] = None
+    ev_min_charge_power_l2: Optional[float] = None
+    ev_min_charge_power_l3: Optional[float] = None
+
+    ev_present_active_power: Optional[float] = None
+    ev_present_active_power_l2: Optional[float] = None
+    ev_present_active_power_l3: Optional[float] = None
+
+    ev_present_reactive_power: Optional[float] = None
+    ev_present_reactive_power_l2: Optional[float] = None
+    ev_present_reactive_power_l3: Optional[float] = None
+
+    ev_max_discharge_power: Optional[float] = None
+    ev_max_discharge_power_l2: Optional[float] = None
+    ev_max_discharge_power_l3: Optional[float] = None
+
+    ev_min_discharge_power: Optional[float] = None
+    ev_min_discharge_power_l2: Optional[float] = None
+    ev_min_discharge_power_l3: Optional[float] = None
+
+    ev_max_v2x_energy_request: Optional[float] = None
+    ev_min_v2x_energy_request: Optional[float] = None
+
+
+@dataclass
+class EVDCCLLimits(Limits):
+    departure_time: Optional[int] = None
+    ev_target_energy_request: Optional[float] = None
+
+    ev_target_current: float = 0.0
+    ev_target_voltage: float = 0.0
+    ev_max_charge_power: Optional[float] = None
+    ev_min_charge_power: Optional[float] = None
+    ev_max_charge_current: Optional[float] = None
+    ev_max_voltage: Optional[float] = None
+    ev_min_voltage: Optional[float] = None
+
+    ev_max_discharge_power: Optional[float] = None
+    ev_min_discharge_power: Optional[float] = None
+    ev_max_discharge_current: Optional[float] = None
+
+    ev_max_energy_request: Optional[float] = None
+    ev_min_energy_request: Optional[float] = None
+
+    ev_max_v2x_energy_request: Optional[float] = None
+    ev_min_v2x_energy_request: Optional[float] = None
+
+
+@dataclass
+class EVACLimits(Limits):
+    ev_max_charge_power: Optional[float] = 0.0
+    ev_max_charge_power_l2: Optional[float] = None
+    ev_max_charge_power_l3: Optional[float] = None
+
+    ev_min_charge_power: Optional[float] = 0.0
+    ev_min_charge_power_l2: Optional[float] = None
+    ev_min_charge_power_l3: Optional[float] = None
+
+    ev_max_discharge_power: Optional[float] = None
+    ev_max_discharge_power_l2: Optional[float] = None
+    ev_max_discharge_power_l3: Optional[float] = None
+    ev_min_discharge_power: Optional[float] = None
+    ev_min_discharge_power_l2: Optional[float] = None
+    ev_min_discharge_power_l3: Optional[float] = None
+
+
+@dataclass
+class EVDCLimits(Limits):
+    ev_max_charge_power: Optional[float] = 0.0
+    ev_min_charge_power: Optional[float] = 0.0
+    ev_max_charge_current: Optional[float] = None
+    ev_min_charge_current: Optional[float] = None
```

Comment:
minor detail: as the CPD comes first than the CL, can we move this class above the the other 2?

also, can we change the names to `EVACCPDLimits` and `EVDCCPDLimits` and add a description where at least the names are expanded?

Reply:
Sure, done.